hy/models.py

```diff
@@ -176,6 +176,18 @@ def __call__(self, data, default=_sentinel):
                 raise
             return default
 
+    # __getstate__ and __setstate__ are required for Pickle protocol
+    # 0, because we have __slots__.
+    def __getstate__(self):
+        return {k: getattr(self, k)
+            for k in (
+                'module', 'start_line', 'end_line', 'start_column',
+                'end_column', 'name')
+            if hasattr(self, k)}
```

Comment:
Instead of fixing those keys, you can use the `__properties__` inherited from `HyObject` and the local `__slots__`, no?

Reply:
Yeah, I think you're right. I'll add an earlier commit to rename `__properties__`, too, because the double underscores on both sides should be reserved for magic names like `__slots__`.